torchvision/transforms/v2/_transform.py

```diff
@@ -1,27 +1,32 @@
 from __future__ import annotations
 
 import enum
-from typing import Any, Callable, Dict, List, Optional, Tuple, Type, Union
+from typing import Any, Callable, Dict, List, Optional, Set, Tuple, Type, Union
 
 import PIL.Image
 import torch
 from torch import nn
 from torch.utils._pytree import tree_flatten, tree_unflatten
 from torchvision import datapoints
+from torchvision.transforms.v2.functional._utils import _get_supported_datapoint_classes
 from torchvision.transforms.v2.utils import check_type, has_any, is_simple_tensor
 from torchvision.utils import _log_api_usage_once
 
 
 class Transform(nn.Module):
-
-    # Class attribute defining transformed types. Other types are passed-through without any transformation
-    # We support both Types and callables that are able to do further checks on the type of the input.
-    _transformed_types: Tuple[Union[Type, Callable[[Any], bool]], ...] = (torch.Tensor, PIL.Image.Image)
+    # This needs to be populated by the subclasses
+    _USED_DISPATCHERS: Set[Type] = {}
 
     def __init__(self) -> None:
         super().__init__()
         _log_api_usage_once(self)
 
+        dispatchers = self._USED_DISPATCHERS.copy()
+        types = _get_supported_datapoint_classes(dispatchers.pop())
+        for dispatcher in dispatchers:
+            types &= _get_supported_datapoint_classes(dispatcher)
+        self._supported_datapoint_types = types
```

Comment:
Is this just

```py
types = {}
for dispatcher in dispatchers:
    types.update(_get_supported_datapoint_classes(dispatcher))
```
?


and `_get_supported_datapoint_classes` probably doesn't need to return a set if we do that

Reply:
I wanted to make sure that we only go for the common ground. Meaning, we only transfer types that are support by _all_ dispatchers. This is why I have the intersection `&` in there.

Reply:
It makes we wonder about the semantic and the use of `_supported_datapoint_types`.. what if there's a transform with multiple dispatchers where each of them handle a different part of the input? Would this transform end-up being a passthrough for everything?

Reply:
I mean typically if we're using the intersection, we can't just have a simple logic in the transforms like

```py
if type(inpt) not in self._supported_datapoint_types:
    # pass-through inpt
```

We can't use the union either because that would pass some inputs to some dispatchers that don't support them (and error).

Seems like we have to implement the pass-through logic on a per-dispatcher basis instead (within the transforms).

Reply:
> Seems like we have to implement the pass-through logic on a per-dispatcher basis instead (within the transforms).

You are right. Let me see how this fits in